packages/discount-code-generator/src/cli.js

```diff
@@ -0,0 +1,217 @@
+import fs from 'fs'
+import npmlog from 'npmlog'
+import PrettyError from 'pretty-error'
+import yargs from 'yargs'
+import json2csv from 'json2csv'
+import csv from 'csv-parser'
+import flatten, { unflatten } from 'flat'
+
+import discountCodeGenerator from './main'
+import { version } from '../package.json'
+
+process.title = 'discountCodeGenerator'
+
+const args = yargs
+  .usage(`
+Usage: $0 [options]
+Generate multiple discount codes to import to the commercetools platform.`,
+  )
+  .showHelpOnFail(false, 'Use --help to display the CLI options.\n')
+
+  .option('help', {
+    alias: 'h',
+  })
+  .help('help', 'Show help text.')
+
+  .option('version', {
+    alias: 'v',
+    type: 'boolean',
+  })
+  .version('version', 'Show version number.', version)
+
+  .option('quantity', {
+    alias: 'q',
+    describe: 'Quantity of discount codes to generate.',
```

Comment:
I tested with various numbers, and calling `JSON.stringify()` causes memory errors on very large numbers. On my system, this limit is about 750,000 (could be more or less on other systems) hence my reason for limiting to 500,000. The alternative to this which I tested would be to stream the output, stringify and write per object using `JSONStream.stringify()`, but this is not feasible as it takes an insane amount of time (writing 1,000,000 codes takes about 20mins). If there's another alternative which I don't know of, I would be happy to be pushed in the right direction, but for now, I think I'll stick to 500,000 codes for optimum performance. The process can be run multiple times if more codes are desired.
Cc @hisabimbola @emmenko